src/openlcb/MultiConfiguredConsumer.hxx

```diff
@@ -61,7 +62,7 @@ public:
         , size_(N)
         , offset_(config)
     {
-        HASSERT(size_ == N);
+        HASSERT(size == N);
```

Comment:
if size is always == N, should we deprecate this constructor and create another one without the size parameter at all?

Reply:
This is a bit of a template trickery to ensure that two arrays are of the same length without the user having to specify either of them. A typical usage is this:
```
openlcb::MultiConfiguredConsumer direct_consumers(stack.node(), kDirectGpio,
    ARRAYSIZE(kDirectGpio), cfg.seg().direct_consumers());
```
Here both N and size are computed by the compiler. Unfortunately we cannot check them statically (without resorting to some ugly macro), but the device will crash deterministically upon first boot if they do not match.

I added documentation comments on this.